3_another_title/README.md

```diff
@@ -0,0 +1,32 @@
+# Answers
+## __new__ and __delete__ operators
+
+### 1. What is a memory leak? How to prevent leaking of memory?
+
+It is a situation when the memory address is no longer accessable. To prevent leaking of memory we have to release the memory after its allocation and usage and before exiting blocks.
+
+### 2. What does happen if allocating memory without providing a pointer?
+
+Memory allocation has * type and can't be converted to basic non-'*' type.
+
+### 3. How to prevent throwing the "bad_alloc" exception? Provide an example.
+
+Exceptions are prevented by mechanisms of exceptions handling: wraping the "dangerous" code with `{try catch}` or simply `throw` an exception.
+
+## Smart pointers
+
+### 1. Why is it impossible to copy one `unique_ptr` to another?
+
+It is imposible because of the nature of `unique_ptr`, that are used to provide unique pointers to objects.
```

Comment:
Is there some way how to assign or move the resource that the one unique pointer owns to another?

Reply:
*that are used to provide unique pointers at current time period. N (N >=2) unique pointers can't point to the same resource simultaneously, but the ownership can be moved (std::move or <unique_ptr>.release()) to another pointer -> pointer, that passed his ownership is no longer containing the address of that resource.